### ISSUE TITLE
Downloading AppImages with zsync fails due to checksum mismatch

### ISSUE THREAD

**Issue by probonopd** [openSUSE/open-build-service#3203]
Issue description
====================

Downloading AppImages generated by OBS from download.opensuse.org using `zsync` fails

Expected result
============

Downloading  using `zsync` works

How to Reproduce
===============

When I want to download an AppImage using `zsync` with

```
wget https://download.opensuse.org/repositories/home:/adrianSuSE:/AppImage/AppImage/QtQuickApp-latest-x86_64.AppImage.zsync
zsync ./QtQuickApp-latest-x86_64.AppImage.zsync -u https://download.opensuse.org/repositories/home:/adrianSuSE:/AppImage/AppImage
```

it fails with

```
failed to retrieve from QtQuickApp-1495317991.fce72f3.glibc2.2.5-x86_64.AppImage
Aborting, download available in QtQuickApp-1495317991.fce72f3.glibc2.2.5-x86_64.AppImage.part
```

Further information
===============

According to @adrianschroeter, OBS does not use `appimagetool -s` to sign AppImages because at the time when appimagetool runs, the signing key is not available for security reasons. Instead, the AppImage is signed after the fact using [obs-sign](https://github.com/openSUSE/obs-sign), which of course changes the AppImage, which means that the zsync file that had been generated previously no longer matches to the payload AppImage file. 

The SHA-1 checksums do not match:

```
wget -c https://download.opensuse.org/repositories/home:/adrianSuSE:/AppImage/AppImage/QtQuickApp-latest-x86_64.AppImage.zsync
wget -c https://download.opensuse.org/repositories/home:/adrianSuSE:/AppImage/AppImage/QtQuickApp-1495317991.fce72f3.glibc2.2.5-x86_64.AppImage

me@host:~$ sha1sum QtQuickApp-1495317991.fce72f3.glibc2.2.5-x86_64.AppImage 
fde5bdf0d0d97ea6279ea618a1381ae234720f57  QtQuickApp-1495317991.fce72f3.glibc2.2.5-x86_64.AppImage

me@host:~$ head QtQuickApp-latest-x86_64.AppImage.zsync -n 8
zsync: 0.6.2
Filename: QtQuickApp-1495317991.fce72f3.glibc2.2.5-x86_64.AppImage
MTime: Wed, 31 May 2017 04:38:08 +0000
Blocksize: 2048
Length: 12944536
Hash-Lengths: 2,2,5
URL: QtQuickApp-1495317991.fce72f3.glibc2.2.5-x86_64.AppImage
SHA-1: 2d05488e0881e661e961b370fde992ae11f72ccd
```

One option around this would be to (re-)generate the zsync file after the AppImage gets signed. Note that merely changing the SHA-1 in the zsync file is likely to be insufficient, since zsync files contain block checksums over the entire payload content.

**Comment by probonopd** [comment]
I can confirm that it is working now with zsync files on OBS generated from now on. Thanks @adrianschroeter.

**Comment by munix9** [comment]
This seems to be broken again?
```
$ wget https://download.opensuse.org/repositories/home:/dmulder:/YaST:/AppImage/AppImage/admin-tools-0-Build27.6.glibc2.15-x86_64.AppImage
$ wget https://download.opensuse.org/repositories/home:/dmulder:/YaST:/AppImage/AppImage/admin-tools-0-Build27.6.glibc2.15-x86_64.AppImage.zsync

$ sha1sum admin-tools-0-Build27.6.glibc2.15-x86_64.AppImage
9c4d44b784d2ff97d13e1c1442798926856d4018  admin-tools-0-Build27.6.glibc2.15-x86_64.AppImage

$ head admin-tools-0-Build27.6.glibc2.15-x86_64.AppImage.zsync -n 8
zsync: 0.6.2
Filename: admin-tools-0-Build27.6.glibc2.15-x86_64.AppImage
MTime: Fri, 18 Oct 2019 02:48:10 +0000
Blocksize: 2048
Length: 64968144
Hash-Lengths: 2,2,5
URL: admin-tools-0-Build27.6.glibc2.15-x86_64.AppImage
SHA-1: 1f81d156b295d7eddfe135b92b3b8fdb47dc5c87
```

**Comment by probonopd** [comment]
cc @TheAssassin, do you know what might be going on here?

**Comment by TheAssassin** [comment]
Nope. One'd have to check how exactly those `.zsync` files are generated.

**Comment by munix9** [comment]
Hi,

after I did some research - I still have a local 2.9 OBS here, but in principle this can also be used on a 2.10 - I noticed the following:

It seems that the package `zsync` with the corresponding program `zsyncmake` is missing on the respective OBS instances on which the signer runs.

I did a test on my OBS instance and without `zsyncmake` the log spits out under `/srv/obs/log/signer.log`:
```
waiting for an event...
regenerating zsync file
Can't exec "zsyncmake": No such file or directory at /usr/lib/obs/server/bs_signer line 470.
zync file recreation failed: -1
```

But if the package `zsync` is installed and a rebuild of the package takes place, then there is
```
regenerating zsync file
```
without errors and the SHA-1 checksums will match again.

Should we open a new issue now or is it enough to open it here again to fix the problem on the OBS (i.e. install the package `zsync` on the signer instances)?
